java2typescript-jackson/src/main/java/java2typescript/jackson/module/grammar/EnumType.java

```diff
@@ -36,16 +37,38 @@ public EnumType(String className) {
 
 	@Override
 	public void writeDefInternal(Writer writer, WriterPreferences preferences) throws IOException {
-		writer.write(format("enum %s {\n", name));
-		preferences.increaseIndentation();
-		if(preferences.isSort()) {
-			Collections.sort(values);
+		if (preferences.isStringLiteralTypeForEnums()) {
```

Comment:
Could You move Your implementation of writing EnumType as String literal to separate class, like I've done with [EnumTypeToEnumPatternWriter](https://github.com/raphaeljolivet/java2typescript/blob/master/java2typescript-jackson/src/main/java/java2typescript/jackson/module/writer/EnumTypeToEnumPatternWriter.java). The idea is to decouple AbstractType and its representation in generated output. (I don't like AbstractType.write* methods, but as i didn't want to break the API for existing users, I didn't get rid of them).